src/mixer/basetrackplayer.cpp

```diff
@@ -343,22 +369,50 @@ void BaseTrackPlayerImpl::slotTrackLoaded(TrackPointer pNewTrack,
                 ConfigKey("[Mixer Profile]", "GainAutoReset"), false)) {
             m_pPreGain->set(1.0);
         }
-        int reset = m_pConfig->getValue<int>(
-                ConfigKey("[Controls]", "SpeedAutoReset"), RESET_PITCH);
-        if (reset == RESET_SPEED || reset == RESET_PITCH_AND_SPEED) {
-            // Avoid resetting speed if master sync is enabled and other decks with sync enabled
-            // are playing, as this would change the speed of already playing decks.
-            if (!m_pEngineMaster->getEngineSync()->otherSyncedPlaying(getGroup())) {
-                if (m_pRateSlider != NULL) {
-                    m_pRateSlider->set(0.0);
+
+        if (!m_cloneFromChannel) {
+            int reset = m_pConfig->getValue<int>(
+                    ConfigKey("[Controls]", "SpeedAutoReset"), RESET_PITCH);
+            if (reset == RESET_SPEED || reset == RESET_PITCH_AND_SPEED) {
+                // Avoid resetting speed if master sync is enabled and other decks with sync enabled
+                // are playing, as this would change the speed of already playing decks.
+                if (!m_pEngineMaster->getEngineSync()->otherSyncedPlaying(getGroup())) {
+                    if (m_pRateSlider != NULL) {
+                        m_pRateSlider->set(0.0);
+                    }
                 }
             }
-        }
-        if (reset == RESET_PITCH || reset == RESET_PITCH_AND_SPEED) {
+            if (reset == RESET_PITCH || reset == RESET_PITCH_AND_SPEED) {
+                if (m_pPitchAdjust != NULL) {
+                    m_pPitchAdjust->set(0.0);
+                }
+            }
+        } else {
+            // perform a clone of the given channel
+
+            // copy rate
+            if (m_pRateSlider != NULL) {
+                m_pRateSlider->set(ControlObject::get(ConfigKey(m_cloneFromChannel->getGroup(), "rate")));
+            }
+
+            // copy pitch
             if (m_pPitchAdjust != NULL) {
-                m_pPitchAdjust->set(0.0);
+                m_pPitchAdjust->set(ControlObject::get(ConfigKey(m_cloneFromChannel->getGroup(), "pitch_adjust")));
+            }
+
+            // copy play state
+            ControlObject::set(ConfigKey(getGroup(), "play"),
+                ControlObject::get(ConfigKey(m_cloneFromChannel->getGroup(), "play")));
```

Comment:
previously I was triggering a play on load, but this caused the track to play before it was seeked and this was audible sometimes. Triggering the play here reduces the gap between play and seek, but it might make sense to move the play operation as part of the `requestSyncPosition()` operation though.

Reply:
I attempted to play from the engine thread as part of the sync operation, but it didn't work quite right.